### ISSUE TITLE
[feature-request] Detect invalid unquoted imports

### ISSUE THREAD

**Issue by smackesey** [astral-sh/ruff#5753]
The following code crashes at runtime:

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from threading import Thread

def foo(t: Thread):
    return t
```

The issue is that `Thread` is not defined at runtime, since it's only imported in a `TYPE_CHECKING` block. But `ruff` does not detect any error.

If this module included `from __future__ import annotations`, then it would be OK because "Thread" would be interpreted as a string-- but the module doesn't include that, so it should be possible to detect that this is illegal.

Further, I think in Python 3.13 string annotations, derived from `from __future__ import annotations` or otherwise, will go away in favor of wrapping annotations in a function for deferred evaluation. See [here](https://lukasz.langa.pl/61df599c-d9d8-4938-868b-36b67fdb4448/) and [PEP 649](https://peps.python.org/pep-0649/) for the dirty details.

**Comment by smackesey** [comment]
Great!

Yes, ^^ is fine because Python never evaluates local variable annotations-- the annotation for a local variable is _always_ a string, regardless of `from __future__ import annotations`.

**Comment by smackesey** [comment]
Btw IMO there should be autofix that converts it to a quoted string-- so my request here is similar to #5559.

**Comment by charliermarsh** [comment]
Unfortunately I've become intimately familiar with all the rules around annotation semantics :joy:

**Comment by charliermarsh** [comment]
Do you think this should just be caught by [F821](https://beta.ruff.rs/docs/rules/undefined-name/), with a custom suggestion and autofix to quote it? Or a separate rule?

**Comment by smackesey** [comment]
Certainly sense to me to catch under an existing undefined anme rule, but I guess ultimately it depends on how much you are trying to mirror exact behavior of the rules you ported from other linters.

**Comment by charliermarsh** [comment]
May be easier to just implement this as the TC200 rules from [flake8-type-checking](https://github.com/snok/flake8-type-checking).

**Comment by charliermarsh** [comment]
This is a bit different than the request in https://github.com/astral-sh/ruff/issues/5559, which is to automatically quote any annotations that _can_ be quoted (and are only used in typing contexts), right?

**Comment by smackesey** [comment]
Yeah it's definitely different, they just both involve adding quotes to annotations.

**Comment by ntBre** [comment]
I think this has been resolved by the combination of [runtime-import-in-type-checking-block (TC004)](https://docs.astral.sh/ruff/rules/runtime-import-in-type-checking-block/#runtime-import-in-type-checking-block-tc004) and the [lint.flake8-type-checking.quote-annotations](https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_quote-annotations) setting.
